mango-core/src/main/scala/org/bdgenomics/mango/converters/AlignmentRecordConverterGA4GH.scala

```diff
@@ -0,0 +1,187 @@
+    builder.setImproperPlacement(!record.getProperPair)
+    builder.setDuplicateFragment(record.getDuplicateRead)
+    builder.setFailedVendorQualityChecks(record.getFailedVendorQualityChecks)
+    builder.setSecondaryAlignment(record.getSecondaryAlignment)
+    builder.setSupplementaryAlignment(record.getSupplementaryAlignment)
+    if (record.getMateContigName != null)
+      builder.setNextMatePosition(ga4gh.Common.Position.newBuilder.setReferenceName(record.getMateContigName)
+        .setPosition(record.getMateAlignmentStart)
+        .setStrand(if (record.getMateNegativeStrand) Common.Strand.NEG_STRAND else Common.Strand.POS_STRAND)
+        .build())
+    // we don't store the number of reads in a fragment; assume 2 if paired, 1 if not
+    val paired: Boolean = Option(record.getReadPaired)
+      .map(b => b: Boolean)
+      .getOrElse(false)
+    val numReads = if (paired) 2 else 1
+    builder.setNumberReads(numReads)
+
+    // however, we do store the read number
+    builder.setReadNumber(record.getReadInFragment)
+
+    // set fragment length
+    Option(record.getInferredInsertSize)
+      .map(l => l.intValue: java.lang.Integer)
+      .foreach((x) => builder.setFragmentLength(x.toInt))
+
+    // set sequence
+    builder.setAlignedSequence(record.getSequence)
+
+    // qual is an array<int> in ga4ghland, so convert
+    // note: avro array<int> --> java.util.List[java.lang.Integer]
+    val qualArray = Option(record.getQual)
+      .map(qual => {
+        qual.toList
+          .map(c => c.toInt - 33)
+      }).getOrElse(List.empty[Int])
+      .map(i => i: java.lang.Integer)
+    builder.addAllAlignedQuality(qualArray)
+
+    // if the read is aligned, we must build a linear alignment
+    Option(record.getReadMapped)
+      .filter(mapped => mapped)
+      .foreach(isMapped => {
+
+        // get us a builder
+        val laBuilder = ga4gh.Reads.LinearAlignment.newBuilder()
+
+        // get values from the ADAM record
+        val start = record.getStart
+        val contig = record.getContigName
+        val reverse: Boolean = record.getReadNegativeStrand
+
+        // check that they are not null
+        require(start != null && contig != null && reverse != null,
+          "Alignment start/contig/strand bad in %s.".format(record))
+
+        // set position
+        laBuilder.setPosition(ga4gh.Common.Position.newBuilder()
+          .setReferenceName(contig)
+          .setPosition(start.intValue)
+          .setStrand(if (reverse) Common.Strand.NEG_STRAND else Common.Strand.POS_STRAND)
+          .build())
+
+        // set mapq
+        laBuilder.setMappingQuality(record.getMapq)
+
+        // convert cigar
+        laBuilder.addAllCigar(convertCigar(record.getCigar))
+
+        // build and attach
+        builder.setAlignment(laBuilder.build)
+      })
+
+    // fin! we skip the info tags for now.
+    builder.build()
+  }
+
+  def listGApbToJson(gaReads: Seq[ga4gh.Reads.ReadAlignment]): String = {
```

Comment:
This naming is a little odd. What is the 'list' for? Is this standard naming that I don't know about?